core/deployment/src/main/java/io/quarkus/deployment/configuration/BooleanConfigType.java

```diff
@@ -50,23 +53,24 @@ public void generateAcceptConfigurationValue(final BytecodeCreator body, final R
     public void acceptConfigurationValueIntoGroup(final Object enclosing, final Field field, final NameIterator name,
             final SmallRyeConfig config) {
         try {
-            field.setBoolean(enclosing, config.getOptionalValue(name.toString(), Boolean.class)
-                    .orElse(Boolean.FALSE).booleanValue());
+            Optional<Boolean> optionalValue = (Optional<Boolean>) ConfigUtils.getOptionalValue(config, name.toString(),
```

Comment:
This is an example of a cast that would be unnecessary by generifying `ConfigUtils#get*Value` (see other comment on that class).